lib/suse/connect/api.rb

```diff
@@ -75,6 +77,27 @@ def forward_system_activations(system)
         )
       end
 
+      def send_bulk_system_update(systems, system_limit = nil)
+        # Test system limit
+        system_limit ||= BULK_SYSTEM_REQUEST_LIMIT
+        last_response = nil
+        systems.each_slice(system_limit) do |batched_systems|
+          params = prepare_payload_for_bulk_update(batched_systems)
+          last_response = make_single_request(
+            :put,
+            "#{connect_api}/organizations/systems",
+            { body: params.to_json }
+          )
+        end
+        last_response
+      rescue RequestError => e
+        # change some params here and start the bulk update.
+        if e.response.code == 413
+          system_limit = e.response.headers['X-Payload-Entities-Max-Limit'].to_i
+          send_bulk_system_update(systems, system_limit)
+        end
```

Comment:
Shouldn't we re-raise the exception if this is not a 413? Otherwise other errors will be discarded, no?

Reply:
Good catch. I will amend this